core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/signer/Checksummer.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.auth.aws.signer;
+
+import java.nio.ByteBuffer;
+import java.util.concurrent.CompletableFuture;
+import org.reactivestreams.Publisher;
+import software.amazon.awssdk.annotations.SdkProtectedApi;
+import software.amazon.awssdk.http.ContentStreamProvider;
+import software.amazon.awssdk.http.auth.aws.internal.signer.DefaultChecksummer;
+
+@SdkProtectedApi
+public interface Checksummer {
+    static Checksummer create() {
+        return new DefaultChecksummer();
+    }
+
+    String checksum(ContentStreamProvider payload);
+
+    CompletableFuture<String> checksum(Publisher<ByteBuffer> payload);
+}
```

Comment:
I like this. I might iterate on it a bit to see if we can make it align with the checksums in the SRA, but I think this is good for this code review.

Can we add javadoc?